src/Microsoft.AspNet.Mvc.Core/ActionResults/ViewExecutor.cs

```diff
@@ -64,23 +63,10 @@ public static class ViewExecutor
 
             response.ContentType = contentTypeHeader.ToString();
 
-            var wrappedStream = new StreamWrapper(response.Body);
-
-            using (var writer = new StreamWriter(wrappedStream, encoding, BufferSize, leaveOpen: true))
+            using (var writer = new HttpResponseStreamWriter(response.Body, encoding))
             {
-                try
-                {
-                    var viewContext = new ViewContext(actionContext, view, viewData, tempData, writer);
-                    await view.RenderAsync(viewContext);
-                }
-                catch
-                {
-                    // Need to prevent writes/flushes on dispose because the StreamWriter will flush even if
```

Comment:
I still have a question regarding this...why is this so much of a concern?...there are couple of unit tests failing which explicitly check this...